internal/controllers/user_devices_controller.go

```diff
@@ -1856,6 +1856,63 @@ func (udc *UserDevicesController) PostMintDevice(c *fiber.Ctx) error {
 	}, sigBytes)
 }
 
+// GetUserDevicesByIntegration godoc
+// @Description Get user devices using integration id
+// @Tags        user-devices
+// @Produce     json
+// @Accept      json
+// @Param       user_device body controllers.GetUserDevicesByIntegration true "all fields are required"
+// @Security    ApiKeyAuth
+// @Success     200 {object} controllers.GetUserDevicesByIntegrationResponse
+// @Security    BearerAuth
+// @Router      /user/devices/:integrationID [post]
+func (udc *UserDevicesController) GetUserDevicesByIntegration(c *fiber.Ctx) error {
+	ctx := context.Background()
```

Comment:
We can use the context of the request So we can clean up if the HTTP request gets canceled.
```suggestion
	ctx := c.Context()
```